### PR TITLE
sessionctx: add tidb_analyze_column_options global variable

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/53567

Problem Summary:
We need to introduce a new global variable to control which columns we need to analyze during the auto-analyze and manual analyze.

Users can set this variable to ALL or PREDICATE to analyze all columns or only predicate columns. The default value will be PREDICATE after this feature is fully implemented.

### What changed and how does it work?

- Added a new global variable called `tidb_analyze_column_options`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] [Manual test](https://github.com/pingcap/tidb/pull/54200#issuecomment-2188438122)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
Tested locally:
1. Start the cluster with the patch: `tiup playground v8.1.0  --db.binpath /Volumes/t7/code/tidb/bin/tide-server`
2. Check the tidb_analyze_default_column_choice:
```sql
mysql> select @@tidb_analyze_default_column_choice;
+--------------------------------------+
| @@tidb_analyze_default_column_choice |
+--------------------------------------+
| ALL                                  |
+--------------------------------------+
1 row in set (0.01 sec)
```
3. Set the tidb_analyze_default_column_choice with lower case value:
```sql
mysql> set global tidb_analyze_default_column_choice = 'predicate'
    -> ;
Query OK, 0 rows affected (0.02 sec)

mysql> select @@tidb_analyze_default_column_choice;
+--------------------------------------+
| @@tidb_analyze_default_column_choice |
+--------------------------------------+
| PREDICATE                            |
+--------------------------------------+
1 row in set (0.00 sec)
```
4. Set the tidb_analyze_default_column_choice with upper case value:
```sql
mysql> set global tidb_analyze_default_column_choice = 'ALL';
Query OK, 0 rows affected (0.01 sec)

mysql> select @@tidb_analyze_default_column_choice;
+--------------------------------------+
| @@tidb_analyze_default_column_choice |
+--------------------------------------+
| ALL                                  |
+--------------------------------------+
1 row in set (0.00 sec)
```
5. Try unknown value:
```sql
mysql> set global tidb_analyze_default_column_choice = 'unkown';
ERROR 1105 (HY000): invalid value for tidb_analyze_default_column_choice, it should be either 'ALL' or 'PREDICATE'
```

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

File: pkg/sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -1372,6 +1377,7 @@ const (
 	DefTiDBMemQuotaAnalyze                         = -1
 	DefTiDBEnableAutoAnalyze                       = true
 	DefTiDBEnableAutoAnalyzePriorityQueue          = true
+	DefTiDBAnalyzeDefaultChoice                    = "ALL"
```
Comment (elsa0520) [inline pkg/sessionctx/variable/tidb_vars.go:1380]:
```suggestion
	DefTiDBAnalyzeDefaultChoice                    = model.AllColumns.String()
```

Comment (hi-rustin) [reply pkg/sessionctx/variable/tidb_vars.go:1380]: We cannot call a function to init the const variable. Awesome Golang!

**Review by elsa0520 — APPROVED** [review]
LGTM

**Review by time-and-fate — APPROVED** [review]

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Pull request merged**